### PR TITLE
[generator] Do not inline duplicate attributes when processing protocols

### PR DESCRIPTION
**Pull request by spouliot**
- this requires small (but tricky) generator changes
- duplicate availability attributes detection is tested by introspection
- it also requires removing dupes in manual bindings (which was the
  current, if not the _named_, goal for this PR)

Long story:

It's always been hard to detect and enforce removal of extraneous
availability attributes since the generator inlined them _freely_
when processing ObjC protocols.

That meant introspection could not report dupes and led to many
extra attributes. This was **not** a problem in binding files since we
could ignore them. IOW their presence (in the input) does not mean they
are _all_ dupes in generated code (output).

However dupes in manual bindings were also (forced to be) _ignored_ and
those were part of the assemblies we ship. Again it was not a big deal,
nor a source of much extra metadata/size, so it was ignored.

Forward to today, manual bindings needs to be updated for net6 [1] so any
extra we have requires more (manual) work. Cleaning this problem up in
the generator code reduce the (manual) work we need to do.

It also means introspection can detect dupes (in generated and manual
code) so we don't end up adding more (which would also require more
manual work to support both set of attributes).

[1] https://github.com/xamarin/xamarin-macios/pull/10580

### CODE REVIEW

**Comment by spouliot** [comment]
Made it a draft PR at the last moment... since I realized it's missing a case to detect dupes on properties (even if the getter and setters are fine).

File: src/VideoToolbox/VTCompressionSession.cs [hunk 1]
```diff
@@ -17,7 +18,13 @@
 using CoreVideo;
 
 namespace VideoToolbox {
+
+#if NET
+	[SupportedOSPlatform ("ios8.0")]
+	[SupportedOSPlatform ("tvos10.2")]
```
Comment (rolfbjarne) [inline src/VideoToolbox/VTCompressionSession.cs:24]: I wonder if we should make these strings constants?

**Review by rolfbjarne — APPROVED** [review]

File: src/VideoToolbox/VTCompressionSession.cs (hunk 1 above)
Comment (spouliot) [reply src/VideoToolbox/VTCompressionSession.cs:24]: hmm... that change was not meant to be included in _this_ PR but yeah I came to the same ~~conclusion~~ question

If we use strings we'll end up with mistakes - even if I added introspection tests against _most_ typos
OTOH using string constants requires a bit more typing
* we can have sharpie generate them...
* but I'd rather only have the old ones inside binding files (and limit the dual set to manual bindings)

Maybe some internal short hands like `_.ios_8_0`...

**Review by dalexsoto — APPROVED** [review]
Looking good! Wish we had the generator diff available

**Comment by spouliot** [comment]
@dalexsoto I'm merging `main` and doing the diff locally. I'll gist the diff of the PR before getting this out of draft :)

**Comment by spouliot** [comment]
generator diff (huge)
https://gist.githubusercontent.com/spouliot/a5d7809e9d8f23a0c1ba0a1d020c442c/raw/8935de7fe803eb6c52852c592a4e41445b3c81ee/gistfile1.txt

**Review by tj-devel709 — APPROVED** [review]

**Pull request merged**